stacker/actions/diff.py

```diff
@@ -0,0 +1,146 @@
+import logging
+
+from .. import exceptions
+from ..plan import COMPLETE, SKIPPED, Plan
+from . import build
+import difflib
+import json
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+class Action(build.Action):
+    """ Responsible for diff'ing CF stacks in AWS and on disk
+
+    Generates the build plan based on stack dependencies (these dependencies
+    are determined automatically based on references to output values from
+    other stacks).
+
+    The plan is then used to pull the current CloudFormation template from
+    AWS and compare it to the generated templated based on the current
+    config.
+    """
+
+    def _diff_parameters(self, old_params, new_params):
+        """Compares the old vs. new parameters and returns a 'diff'
+
+        Args:
+            old_params(dict): old paramters
+            new_params(dict): new parameters
+
+        Returns:
+            int: number of changed entries
+            string: "diff" of all parameters and how they changed
+        """
+        output = []
```

Comment:
You might be able to make this logic a little easier if you use sets on the keys - an example:

```
>>> old = {'a': 'Apple', 'b': 'Banana', 'c': 'Carrot'}
>>> new = {'b': 'Banana', 'a': 'Apple', 'o': 'Orange', 'l': 'Lettuce'}
>>>
>>> old_set = set(old)
>>> new_set = set(new)
>>>
>>> added = new_set - old_set
>>> removed = old_set - new_set
>>> common = old_set & new_set
>>>
>>> added
set(['l', 'o'])
>>> removed
set(['c'])
>>> common
set(['a', 'b'])
```

Then you can iterate over the common ones to see if the values are actually different or not.  The other thing this gives you is the ability to print the removed values and not rely on a special string like `_Undef_`